src/sessions/p2p_session.rs

```diff
@@ -140,6 +142,11 @@ where
     event_queue: VecDeque<GGRSEvent<T>>,
     /// Contains all local inputs not yet sent into the system. This should have inputs for every local player before calling advance_frame
     local_inputs: HashMap<PlayerHandle, PlayerInput<T::Input>>,
+
+    /// With desync detection, the session will compare checksums for all peers to detect discrepancies / desyncs between peers
+    desync_detection: DesyncDetection,
+    /// Desync detection over the network
+    checksum_history: HashMap<Frame, u128>,
```

Comment:
If this one is the history of own checksums and `UdpProtocol.checksum_history` is the history of remote checksums for that endpoint, we should probably name this one `local_checksum_history` to distinguish them.